net/FlatBuffers/ByteBuffer.cs

```diff
@@ -889,7 +921,65 @@ public int Put<T>(int offset, T[] x)
 #if ENABLE_SPAN_T && (UNSAFE_BYTEBUFFER || NETSTANDARD2_1)
                 MemoryMarshal.Cast<T, byte>(x).CopyTo(_buffer.Span.Slice(offset, numBytes));
 #else
-                Buffer.BlockCopy(x, 0, _buffer.Buffer, offset, numBytes);
+                var srcOffset = ByteBuffer.SizeOf<T>() * x.Offset;
+                Buffer.BlockCopy(x.Array, srcOffset, _buffer.Buffer, offset, numBytes);
+#endif
+            }
+            else
+            {
+                throw new NotImplementedException("Big Endian Support not implemented yet " +
+                    "for putting typed arrays");
+                // if we are BE, we have to swap each element by itself
+                //for(int i = x.Length - 1; i >= 0; i--)
+                //{
+                //  todo: low priority, but need to genericize the Put<T>() functions
+                //}
+            }
+            return offset;
+        }
+
+        /// <summary>
+        /// Copies an array segment of type T into this buffer, ending at the given
+        /// offset into this buffer. The starting offset is calculated based on the count
+        /// of the array segment and is the value returned.
+        /// </summary>
+        /// <typeparam name="T">The type of the input data (must be a struct)</typeparam>
+        /// <param name="offset">The offset into this buffer where the copy will end</param>
+        /// <param name="ptr">The pointer to copy data from</param>
+        /// <param name="sizeInBytes">The number of bytes to copy</param>
+        /// <returns>The 'start' location of this buffer now, after the copy completed</returns>
+        public int Put<T>(int offset, IntPtr ptr, int sizeInBytes)
```

Comment:
Maybe name this ending_offset to clarify that? same with the other overloads.

Reply:
I'd like to keep the naming consistent within the file. The 'offset' parameter in the currently existing overloads are named 'offset' as well (e.g. Put<T>(int offset, T[] x)).

I'd suggest to either rename ALL of these parameters ending_offset or to keep 'offset'. Up to you :).

Reply:
Ok, I think it is fine to leave it for now.